python/lancedb/table.py

```diff
@@ -587,11 +590,23 @@ def create_fts_index(self, field_names: Union[str, List[str]]):
         ----------
         field_names: str or list of str
             The name(s) of the field to index.
+        replace: bool, default False
+            If True, replace the existing index if it exists. Note that this is
+            not yet an atomic operation; the index will be temporarily
+            unavailable while the new index is being created.
         """
         from .fts import create_index, populate_index
 
         if isinstance(field_names, str):
             field_names = [field_names]
+
+        fs, path = fs_from_uri(self._get_fts_index_path())
+        index_exists = fs.get_file_info(path).type != pa_fs.FileType.NotFound
+        if index_exists and not replace:
+            raise ValueError(f"Index already exists. Use replace=True to overwrite.")
+        elif index_exists:
+            fs.delete_dir(path)
```

Comment:
this is super nit-picky but wdyt about:

```suggestion
        if index_exists: 
            if not replace:
                raise ValueError(f"Index already exists. Use replace=True to overwrite.")
            fs.delete_dir(path)
```